OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/CallGraphAnalysis.scala

```diff
@@ -121,25 +120,27 @@ class CallGraphAnalysis private[cg] (
 
     override final def processMethod(
         callContext: ContextType,
-        tacEP:       EPS[Method, TACAI]
+        tacEPOpt:    Option[EPS[Method, TACAI]]
     ): ProperPropertyComputationResult = {
-        val state = new CGState[ContextType](callContext, tacEP)
-        if (tacEP ne null)
+        if (tacEPOpt.isDefined) {
+            val state = new TACAIBasedCGState[ContextType](callContext, tacEPOpt.get)
             processMethod(state, new DirectCalls())
-        else
-            returnResult(new DirectCalls(), enforceCalleesResult = true)(state)
+        } else {
+            val state = new CGState[ContextType](callContext)
+            Results {
+                new DirectCalls().partialResults(state.callContext, enforceCalleesResult = true)
```

Comment:
Does this actually do anything? Doesn't look like it.
I think this produces no results at all while missing what the original `returnResults` call would have returned: the InterimPartialResult for the state's dependees.

Reply:
It does do something: Enforce a partial callees result to be produced as it would by an empty direct calls helper instance. If you look at the previous code, it was simply calling `returnResult` with the state and a newly instantiated helper object. Since the state cannot have a dependee since we did not provide it any (and it does not have access to the project or propertyStore), no `InterimResult` could be ever produced. Thus, we can specialize `returnResult` to use the TACAI-Based state to use in the continuation and return our result independently here.

I will make this clearer by removing the newly created state entirely, since we only use the `callContext` from it anyways.

Reply:
Oh, right, I didn't see that.